src/mame/namco/namcos1_m.cpp

```diff
@@ -996,39 +1018,49 @@ u8 namcos1_state::faceoff_inputs_r(offs_t offset)
 		   much time reading the inputs and won't have enough cycles to play two
 		   digital sounds at once. This value is enough to read all inputs at least
 		   once per frame */
-		if (++m_strobe_count > 8)
+		u8 strobe_count = m_strobe_count + 1;
+		u8 input_count = m_input_count;
+		u8 stored_input[2] = {m_stored_input[0], m_stored_input[1]};
+		if (strobe_count > 8)
 		{
-			m_strobe_count = 0;
+			strobe_count = 0;
 
-			res |= m_input_count;
+			res |= input_count;
 
-			switch (m_input_count)
+			switch (input_count)
 			{
 				case 0:
-					m_stored_input[0] = m_io_in[0]->read() & 0x1f;
-					m_stored_input[1] = (m_io_in[3]->read() & 0x07) << 3;
+					stored_input[0] = m_io_in[0]->read() & 0x1f;
+					stored_input[1] = (m_io_in[3]->read() & 0x07) << 3;
 					break;
 
 				case 3:
-					m_stored_input[0] = m_io_in[2]->read() & 0x1f;
+					stored_input[0] = m_io_in[2]->read() & 0x1f;
 					break;
 
 				case 4:
-					m_stored_input[0] = m_io_in[1]->read() & 0x1f;
-					m_stored_input[1] = m_io_in[3]->read() & 0x18;
+					stored_input[0] = m_io_in[1]->read() & 0x1f;
+					stored_input[1] = m_io_in[3]->read() & 0x18;
 					break;
 
 				default:
-					m_stored_input[0] = 0x1f;
-					m_stored_input[1] = 0x1f;
+					stored_input[0] = 0x1f;
+					stored_input[1] = 0x1f;
 					break;
 			}
```

Comment:
You don’t need the local `stored_input` – you can skip this switch statement entirely if side effects are disabled since the values of the stored inputs aren’t used for calculating the `res` value in this path of the `if (strobe_count > 8)`.